File: model/store.go
package model

import (
	"DineEasy/utils/errmsg"
)

// Store 门店表
type Store struct {
	Model
	Name      string `gorm:"comment:门店名字;not null;unique;"  json:"name"`
	Address   string `gorm:"comment:门店地址;"  json:"address"`
	Phone     string `gorm:"comment:门店电话;"  json:"phone"`
	Picture   string `gorm:"comment:门店照片;"  json:"picture"`
	StartTime uint   `gorm:"comment:营业开始时间;" json:"start_time"`
	EndTime   uint   `gorm:"comment:营业结束时间;" json:"end_time"`
	Status    uint   `gorm:"comment:门店状态;default:0;not null;"  json:"status"`
}

// AddStore 增加门店
func AddStore(data *Store) (int, uint) {
	err := Db.Create(data).Error
	if err != nil {
		return errmsg.ERROR, 0
	}
	return errmsg.SUCCESS, data.ID
}

// GetStores 获取门店列表
func GetStores(pageNum, pageSize int) (int, *[]Store, int64) {
	var data []Store
	var count int64
	//根据偏移量获取数据
	err := Db.Offset((pageNum - 1) * pageSize).Limit(pageSize).Find(&data).Error
	if err != nil {
		return errmsg.ERROR, nil, 0
	}
	// 获取总数
	Db.Model(&Store{}).Count(&count)
	return errmsg.SUCCESS, &data, count
}

File: api/v1/Store.js
import Request from "@/api/request.js"
let request = new Request().http


/***
 * @description: 获取门店
 * @param . pagenum:偏移量, pagesize:数量
 */
export function getStores(pagenum, pagesize) {
  return request({
    url: "/api/v1/stores",
    method: "GET",
    data: {
      pagenum,
      pagesize
    }
  })
}


/***
 * @description: 增加门店
 * @param . name:门店名字,address:门店地址,phone:门店电话,picture:门店照片,startTime:营业开始时间,endTime:营业结束时间,status:门店状态
 */
export function addStore(data) {
  return request({
    url: "/api/v1/stores",
    method: "POST",
    data
  })
}
